File: Product of Array Except Self.cpp
class Solution
{
public:
    vector<int> productExceptSelf(vector<int> &nums)
    {
        int n = 1;
        vector<int> ans(nums.size());
        for (int i = 0; i < ans.size(); ++i)
        {
            ans[i] = n;
            n *= nums[i];
        }
        n = 1;
        for (int i = nums.size() - 1; i >= 0; --i)
        {
            ans[i] *= n;
            n *= nums[i];
        }

        return ans;
    }
};

File: product-of-array-except-self/product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        num = 1
        ans = [0] * len(nums)
        for idx, i in enumerate(nums):
            ans[idx] = num
            num *= i
        
        num = 1
        for i in range(len(nums) - 1, -1, -1):
            ans[i] *= num
            num *= nums[i]
        return ans